src/ringct/rctSigs.cpp

```diff
@@ -575,7 +610,10 @@ namespace rct {
         rv.type = RCTTypeFull;
```

Comment:
I believe this should be:

        rv.type = bulletproof ? RCTTypeFullBulletproof : RCTTypeFull;